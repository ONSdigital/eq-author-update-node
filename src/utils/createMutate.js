const createMutate = (client, mutation) => (options) =>
  client.mutate({
    mutation,
    ...options,
  });

export default createMutate;
