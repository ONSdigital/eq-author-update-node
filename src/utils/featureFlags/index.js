// import config from "config";

export const enableOn = (flags) => {
  // const enabledFlags = config.REACT_APP_FEATURE_FLAGS.split(",");

  // const displayFeature = flags.every((flag) => enabledFlags.includes(flag));

  // return displayFeature;

  return true;
};

export const disableOn = (flags) => !enableOn(flags);
