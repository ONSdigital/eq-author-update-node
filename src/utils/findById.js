import { find } from "lodash";
const findById = (collection, id) => find(collection, { id });

export default findById;
