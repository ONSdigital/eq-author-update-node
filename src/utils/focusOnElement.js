const focusOnElement = (id) => document.getElementById(id).focus();

export default focusOnElement;
