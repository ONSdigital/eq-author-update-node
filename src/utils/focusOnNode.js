const focusOnNode = (ref) => {
  if (ref && ref.current) {
    ref.current.focus();
  }
};

export default focusOnNode;
