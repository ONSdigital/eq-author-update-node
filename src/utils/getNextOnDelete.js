import { findIndex } from "lodash";

const getNextOnDelete = (items, id) => {
  const index = findIndex(items, { id });
  let nextIndex;

  if (index === 0) {
    nextIndex = items.length > 1 ? 1 : 0;
  } else {
    nextIndex = index - 1;
  }

  return items[nextIndex];
};

export default getNextOnDelete;
