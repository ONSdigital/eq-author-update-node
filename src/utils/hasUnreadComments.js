/*
  comments: Array of comments from the page
  userId: ID of the user currently signed in (from MeContext)
*/
const hasUnreadComments = (comments, userId) => {
  let hasNotRead = false;
  if (!comments) {
    hasNotRead = false;
    return hasNotRead;
  }
  comments.forEach((comment) => {
    // If signed in user has not read the comment
    if (!comment.readBy.some((id) => userId === id)) {
      hasNotRead = true;
    }
    comment.replies.forEach((reply) => {
      // If signed in user has not read the reply
      if (!reply.readBy.some((id) => userId === id)) {
        hasNotRead = true;
      }
    });
  });
  return hasNotRead;
};

export default hasUnreadComments;
