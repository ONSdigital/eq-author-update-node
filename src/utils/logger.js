/* eslint-disable import/unambiguous */

let logger;

switch (process.env.NODE_ENV) {
  case "development": {
    logger = require("pino")({
      prettyPrint: {
        colorize: true,
        ignore: "pid,hostname",
        translateTime: "dd/mm/yyyy h:MM:ss",
      },
    });
    break;
  }
  default: {
    logger = require("pino")();
  }
}

module.exports = {
  logger,
};
