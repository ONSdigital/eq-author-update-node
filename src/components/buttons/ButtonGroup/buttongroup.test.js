import React from "react";
import ButtonGroup from ".";
import { render } from "tests/utils/rtl";

describe("button group component", () => {
  describe("snapshot", () => {
    it("Should not have changed inadvertently", () => {
      expect(render(<ButtonGroup />).asFragment()).toMatchSnapshot();
    });
  });

  describe("orientation", () => {
    it("is vertical by default", () => {
      const { getByTestId } = render(<ButtonGroup />);
      expect(getByTestId("button-group")).toHaveStyleRule(
        "flex-direction",
        "column"
      );
    });
    it("can be horizontal", () => {
      const { getByTestId } = render(<ButtonGroup horizontal />);
      expect(getByTestId("button-group")).toHaveStyleRule(
        "flex-direction",
        "row"
      );
    });
  });
});
