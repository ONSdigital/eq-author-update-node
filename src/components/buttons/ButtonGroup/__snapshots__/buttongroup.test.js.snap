// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button group component snapshot Should not have changed inadvertently 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
    class="c0"
    data-test="button-group"
  />
</DocumentFragment>
`;
